app/service/FileService.scala

```diff
@@ -55,6 +56,19 @@ class FileService @Inject()(connector: BindingTariffFilestoreConnector) {
     ).map(_.filter(_.isDefined).map(_.get))
   }
 
+  def validate(file: MultipartFormData.FilePart[TemporaryFile]): Either[String, MultipartFormData.FilePart[TemporaryFile]] = {
+
+    file match {
+      case f if hasWrongSize (f) => Left(messagesApi("uploadWrittenAuthorisation.error.size"))
+      case f if hasWrongContentType (f) => Left(messagesApi("uploadWrittenAuthorisation.error.fileType"))
+      case _ => Right(file)
+    }
+
+  }
+  private def hasWrongSize : MultipartFormData.FilePart[TemporaryFile] => Boolean = { file  => file.ref.file.length() > maxFileSize}
+  private def hasWrongContentType : MultipartFormData.FilePart[TemporaryFile] => Boolean = {file  => file.contentType.filter(allowedTypes.contains).isEmpty}
```

Comment:
can be improved as 
```
{ _.contentType.filter(allowedTypes.contains).isEmpty }
```

But I would say:
```
!allowedTypes.contains(_.contentType)
```

Reply:
ContentType is an Option[String] so the allowedTypes won't contain it.  But I'll refactor both function definitions as you suggest